connector/src/main/java/org/jboss/as/connector/subsystems/jca/JcaWorkManagerDefinition.java

```diff
@@ -71,11 +85,57 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
 
     @Override
         public void registerChildren(ManagementResourceRegistration resourceRegistration) {
-        resourceRegistration.registerSubModel(BoundedQueueThreadPoolResourceDefinition.create(WORKMANAGER_SHORT_RUNNING, ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, ThreadsServices.STANDARD_HANDOFF_EXECUTOR_RESOLVER,
-                ThreadsServices.EXECUTOR.append(WORKMANAGER_SHORT_RUNNING), registerRuntimeOnly));
-        resourceRegistration.registerSubModel(BoundedQueueThreadPoolResourceDefinition.create(WORKMANAGER_LONG_RUNNING, ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, ThreadsServices.STANDARD_HANDOFF_EXECUTOR_RESOLVER,
-                        ThreadsServices.EXECUTOR.append(WORKMANAGER_LONG_RUNNING), registerRuntimeOnly));
+        registerSubModels(resourceRegistration, registerRuntimeOnly);
+    }
+
+    static void registerSubModels(ManagementResourceRegistration resourceRegistration, boolean runtimeOnly) {
+        final BoundedQueueThreadPoolAdd shortRunningThreadPoolAdd = new BoundedQueueThreadPoolAdd(false,
+                ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, ThreadsServices.STANDARD_HANDOFF_EXECUTOR_RESOLVER,
+                ThreadsServices.EXECUTOR.append(WORKMANAGER_SHORT_RUNNING)) {
+            @Override
+            protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+                    throws OperationFailedException {
+                checkThreadPool(context, operation, WORKMANAGER_SHORT_RUNNING);
+                super.populateModel(context, operation, resource);
+            }
+        };
+        resourceRegistration.registerSubModel(
+                new JCAThreadPoolResourceDefinition(false, runtimeOnly, WORKMANAGER_SHORT_RUNNING, ThreadsServices.EXECUTOR.append(WORKMANAGER_SHORT_RUNNING),
+                        CommonAttributes.BOUNDED_QUEUE_THREAD_POOL, shortRunningThreadPoolAdd, new BoundedQueueThreadPoolRemove(shortRunningThreadPoolAdd)));
+
+        final BoundedQueueThreadPoolAdd longRunningThreadPoolAdd = new BoundedQueueThreadPoolAdd(false,
+                ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, ThreadsServices.STANDARD_HANDOFF_EXECUTOR_RESOLVER,
+                ThreadsServices.EXECUTOR.append(WORKMANAGER_LONG_RUNNING)) {
+            @Override
+            protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+                    throws OperationFailedException {
+                checkThreadPool(context, operation, WORKMANAGER_LONG_RUNNING);
```

Comment:
@ctomc IMHO, this can't be done in a `ParameterValidator`, because the `ParameterValidator` can only validate the value of the attribute itself I think. In this case, we need to validate whether there is one ThreadPool with same type(long-running-threads or short-running-threads) defined already under the same `WorkManager`, which is kind of context related.